sequencing-server/src/lib/filters/utilities.ts

```diff
@@ -0,0 +1,272 @@
+    let included = !anyMainFiltersSpecified;
+
+    // Check if directive fits in time range
+    // Currently, it only disqualifies if the start is out of range. If the end is out of range, it is included.
+    if (simulatedActivity.endTime) {
+        if (simulatedActivity.startTime.epochMicroseconds < timeRangeStart.epochMicroseconds ||
+            simulatedActivity.startTime.epochMicroseconds > timeRangeEnd.epochMicroseconds
+        ) {
+            return false;
+        }
+    }
+    else {
+        if (simulatedActivity.startTime.epochMicroseconds < timeRangeStart.epochMicroseconds) {
+            return false;
+        }
+    }
+
+    // Check to see if directive is included in static list
+    if (filter.static_types?.length) {
+        included = !!staticTypeMap[simulatedActivity.activityTypeName];
+    }
+
+    // Check if necessary to see if directive is included in dynamic list
+    if ((!filter.static_types?.length || !included) && filter.dynamic_type_filters?.length) {
+        included = directiveOrSpanMatchesDynamicFilters(
+            simulatedActivity,
+            filter.dynamic_type_filters,
+        );
+    }
+
+    // Apply other filters on top of the types
+    if (filter.other_filters?.length) {
+        included =
+            directiveOrSpanMatchesDynamicFilters(simulatedActivity, filter.other_filters) &&
+            (anyTypeFiltersSpecified ? included : true);
+    }
+
+    // Apply type specific filters if found and if the type is already included or
+    // if no other filters were specified (case where all types are included by default)
+    if (
+        filter.type_subfilters &&
+        filter.type_subfilters[simulatedActivity.activityTypeName] &&
+        filter.type_subfilters[simulatedActivity.activityTypeName]?.length
+    ) {
+        included =
+            directiveOrSpanMatchesDynamicFilters(
+                simulatedActivity,
+                filter.type_subfilters[simulatedActivity.activityTypeName] ?? [], // we know it shouldn't be undefined, though.
+            ) && (anyMainFiltersSpecified ? included : true);
+    }
+    return included;
+}
+
+function directiveOrSpanMatchesDynamicFilters(
+    simulatedActivity: SimulatedActivity<Record<string, unknown>, Record<string, unknown>>,
+    dynamicFilters: ActivityLayerDynamicFilter<typeof ActivityLayerFilterField>[],
+    // activityTypeDefMap: Record<string, ActivityType>,
+    // defaultArgumentsMap: DefaultEffectiveArgumentsMap,
+): boolean {
+    return dynamicFilters.reduce((acc, curr) => {
+        let matches = false;
+        if (curr.field === 'Type') {
+            matches = matchesDynamicFilter(simulatedActivity.activityTypeName, curr.operator, curr.value);
+        } else if (curr.field === 'Name') {
+            // TODO: is id correct?
+            matches = matchesDynamicFilter(simulatedActivity.id, curr.operator, curr.value);
+        // } else if (curr.field === 'Subsystem') {
```

Comment:
Large blocks of commented-out code here. Are they present in the original, or is this a deviation?

Reply:
They can be removed.